deindex.py

```diff
@@ -0,0 +1,167 @@
+                if items == []:
+                    delete_list.append(list_name)
+            for list_name in delete_list:
+                item_details.pop(list_name)
+            if item_details == {}:
+                delete_site.append(site_url)
+        for site_url in delete_site:
+            lists_item_ids.pop(site_url)
+        return ids
+
+    def deindexing_lists(self, collection, ids):
+        """Fetches the id's of deleted lists from the sharepoint server and
+            further removes them from workplace search by invoking the delete
+            document api.
+            Returns:
+                sites: list of site paths
+        """
+        logger.info("Deindexing lists...")
+        lists_ids = ids.get('lists')
+        delete = []
+        for site_url, list_details in lists_ids.items():
+            doc = []
+            for list_id, list_name in list_details.items():
+                url = f"{self.sharepoint_host}{site_url}/_api/web/lists/getbytitle(\'{list_name}\')"
+                resp = self.sharepoint_client.get(url, '?')
+                if resp.status_code == requests.codes['not_found']:
+                    doc.append(list_id)
+            self.ws_client.delete_documents(
+                http_auth=self.ws_token,
+                content_source_id=self.ws_source,
+                document_ids=doc)
+            for id in doc:
+                list_details.pop(id)
+            if list_details == {}:
+                delete.append(site_url)
+        for site_url in delete:
+            lists_ids.pop(site_url)
+        return ids
+
+    def deindexing_sites(self, collection, ids):
+        """Fetches the ids' of deleted sites from the sharepoint server and
+            invokes delete documents api for those ids to remove them from
+            workplace search
+        """
+
+        logger.info("Deindexing sites...")
+
+        site_details = ids.get("sites")
+        doc = []
+        for site_id, site_url in site_details.items():
+            url = f"{self.sharepoint_host}{site_url}/_api/web"
+            resp = self.sharepoint_client.get(url, '?')
+            if resp.status_code == requests.codes['not_found']:
+                doc.append(site_id)
+        self.ws_client.delete_documents(
+            http_auth=self.ws_token,
+            content_source_id=self.ws_source,
+            document_ids=doc)
+        for id in doc:
+            site_details.pop(id)
+        return ids
+
+
+def start():
+    """Runs the de-indexing logic regularly after a given interval
+        or puts the connector to sleep
+    """
+    logger.info('Starting the de-indexing..')
+    config = Configuration("sharepoint_connector_config.yml", logger)
+
+    if not config.validate():
+        print_and_log(
+            logger, 'error', 'Terminating the de-indexing as the configuration parameters are not valid')
+        exit(0)
+    deindexing_interval = 60
+    while True:
+        data = config.reload_configs()
```

Comment:
If we reload the config, we should also re-validate it.
If the config is not valid, it would be better to exit the process altogether.
Also I have to ask, do we need to reload the config file here?
It would be okay to ask operators to restart the process if they want to modify the config.